pgosupport.cmake

```diff
@@ -0,0 +1,80 @@
+function(clr_pgo_unknown_arch)
+    if (WIN32)
+        message(FATAL_ERROR "Only AMD64, ARM and I386 are supported for PGO")
+    else()
+        message(FATAL_ERROR "PGO not currently supported on the current platform")
+    endif()
+endfunction(clr_pgo_unknown_arch)
+
+# Adds Profile Guided Optimization (PGO) flags to the current target
+function(add_pgo TargetName)
+    if(WIN32)
+        set(ProfileFileName "${TargetName}.pgd")
+    endif(WIN32)
+
+    file(TO_NATIVE_PATH
+        "${CLR_CMAKE_PACKAGES_DIR}/${CLR_CMAKE_OPTDATA_PACKAGEWITHRID}/${CLR_CMAKE_OPTDATA_VERSION}/data/${ProfileFileName}"
+        ProfilePath
+    )
+
+    # Enable PGO only for optimized configs
+    set(ConfigTypeList RELEASE RELWITHDEBINFO)
```

Comment:
This would have no effect on single config generators like Unix make generator. In that case, there is just single LINK_FLAGS property, not LINK_FLAGS_xxx per each config. 
So for Unix, you'll need to use if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE OR UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO) and add to LINK_FLAGS if it is true.
So I would just put the whole body of the add_pgo under if(WIN32) for now.